app/javascript/components/DecisionTree.jsx

```diff
@@ -0,0 +1,290 @@
+import React, { Component } from 'react'
+
+import Title from '../ui/Title'
+import Headline from '../ui/Headline'
+import Field from '../ui/Field'
+import uuidv4 from 'uuid/v4'
+
+import {
+  Button,
+  Paper,
+  TextField
+} from 'react-md'
+
+import RelevanceField from './RelevanceField'
+
+export default class DecisionTree extends Component {
+
+  render() {
+    const { skill, actions } = this.props
+    const { name, config } = skill
+
+    const updateConfig = (key) => {
+      return (value) => {
+        actions.updateSkill({
+          ...skill,
+          config: {
+            ...skill.config,
+            [key]: value
+          }
+        })
+      }
+    }
+
+    return (
+      <div>
+        <Title>Decision tree</Title>
+        <Headline>
+          This skill will guide the user for diagnostic purposes or information access
+        </Headline>
+
+        <RelevanceField value={config.relevant} onChange={updateConfig('relevant')} />
+
+        <Field id="tree-explanation" label="Skill explanation"
+               value={config.explanation} onChange={updateConfig('explanation')} />
+        <Field id="tree-clarification" label="Skill clarification"
+               value={config.clarification} onChange={updateConfig('clarification')} />
+        <Field id="tree-clarification" label="Valid keywords (comma separated)"
+          value={config.keywords} onChange={updateConfig('keywords')} />
+        <DecisionTreeComponent tree={config.tree} onChange={updateConfig('tree')} />
+      </div>
+    )
+  }
+}
+
+class DecisionTreeComponent extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      path: this._buildDefaultPath(props.tree.nodes, props.tree.initial),
+      editingOptionTo: null
+    }
+  }
+
+  _buildDefaultPath(nodes, nodeId) {
+    let result = [nodeId]
+    let defaultOption = nodes[nodeId].options[0]
+    while (defaultOption) {
+      const nextId = defaultOption.next
+      result.push(nextId)
+      defaultOption = nodes[nextId].options[0]
+    }
+    return result
+  }
+
+  _deleteOption(nodes, node, optionIx) {
+
+    //remove option
+    nodes = {
+      ...nodes,
+      [node.id]: {
+        ...node,
+        options: [
+          ...node.options.slice(0, optionIx),
+          ...node.options.slice(optionIx + 1)
+        ]
+      }
+    }
+
+    const removeNode = function(nodeId) {
+      const currentNode = nodes[nodeId]
+      currentNode.options.forEach((option, ix) => {
+        removeNode(option.next)
+      })
+      delete nodes[nodeId]
+    }
+
+    // get node referenced by option
+    const nextId = node.options[optionIx].next
+    removeNode(nextId)
+
+    this.setState({
+      path: this._buildDefaultPath(nodes, this.state.path[0])
```

Comment:
This make the default path to always be the left most path after deleting *any* option, right? Should preserve the path until the level at least.

Ideally I would select the following option in that level (if exists)